tests/com.b2international.snowowl.test.commons/src/com/b2international/snowowl/test/commons/codesystem/CodeSystemVersionRestRequests.java

```diff
@@ -108,34 +107,24 @@ public static ValidatableResponse getVersions(String shortName) {
 		}
 	}
 
-	public static Date getNextAvailableEffectiveDate(String shortName) {
-		SortedSet<String> effectiveDates = getEffectiveDates(shortName);
-		Calendar calendar = Calendar.getInstance();
+	public static LocalDate getNextAvailableEffectiveDate(String shortName) {
+		LocalDate effectiveDate = LocalDate.now();
 
+		SortedSet<String> effectiveDates = getEffectiveDates(shortName);
 		if (!effectiveDates.isEmpty()) {
-			Date latestEffectiveDate = Dates.parse(effectiveDates.last(), DateFormats.SHORT);
+			LocalDate latestEffectiveDate = EffectiveTimes.parse(effectiveDates.last(), DateFormats.SHORT);
 			// XXX make sure we always use today or later dates, so all versions created in chronological order, even if some of the pre-imported content we are relying on are still in the past
 			// and adding one day to that historical version would mean a historical effective time version, which is unfortunate and can lead to inconsistencies in tests
```

Comment:
The removed parts snapped `latestEffectiveDate` to today's date if it is in the past; the updated code starts with today's date and snaps to the latest effective date if it is in the future 😄 The comment should be updated and moved to the top of the method, since the process now starts there.

Reply:
Hmm, I'll check again, thanks. Also, I'll remove the same functionality from other test cases in other projects.